flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/watermarkstatus/HeapPriorityQueue.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.watermarkstatus;
+
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+
+import static org.apache.flink.util.CollectionUtil.MAX_ARRAY_SIZE;
+
+/**
+ * This class is similar to the {@link org.apache.flink.runtime.state.heap.HeapPriorityQueue}, which
+ * has the similar functions with {@link org.apache.flink.runtime.state.heap.HeapPriorityQueue}.
+ * This class is introduced as the replacement of {@link
+ * org.apache.flink.runtime.state.heap.HeapPriorityQueue} to be used in {@link
+ * StatusWatermarkValve}, to avoid affecting the performance of memory state backend.
```

Comment:
The statement is a bit redundant. Maybe simplify it a bit.

```suggestion
 * This class has similar functions with {@link org.apache.flink.runtime.state.heap.HeapPriorityQueue}.
 * It is introduced as the replacement of {@link
 * org.apache.flink.runtime.state.heap.HeapPriorityQueue} to be used in {@link
 * StatusWatermarkValve}, to avoid affecting the performance of memory state backend.
```